docs/source/openapi.yml

```diff
@@ -0,0 +1,198 @@
+                      type: string
+                    new:
+                      type: object
+      responses:
+        '400':
+          description: Invalid ID supplied
+        '404':
+          description: Sample not found
+        '405':
+          description: Validation exception
+  '/samples/{uuid}/revisions':
+    get:
+      tags:
+        - samples
+      summary: Find revisions of a sample by uuid
+      description: Returns revisons of a sample with specific uuid
+      operationId: getSampleRevisionsByuuid
+      parameters:
+        - name: uuid
+          in: path
+          description: Globally unique ID
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  revisons:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/Sample'
+        '400':
+          description: Invalid uuid supplied
+        '404':
+          description: Revisons not found
+  /samples/new:
+    post:
+      tags:
+        - samples
+      summary: create a new sample
+      description: ''
+      operationId: newSample
+      requestBody:
+        description: Optional description in *Markdown*
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                name:
+                  type: string
+                  description: Sample name
+                projects:
+                  description: List of foreign keys to Projects
+                  type: array
+                  items:
+                    type: string
+                composition:
+                  type: string
+                  description: Sample composition (IUCr specification)
+                tags:
+                  type: array
+                  items:
+                    type: string
+                  description: Arbitrary useful labels
+                description:
+                  type: string
+                  description: Longer sample description
+              required:
+                - name
+      responses:
+        '200':
+          description: successful operation
+        '400':
+          description: Invalid ID supplied
+  /samples:
+    post:
+      tags:
+        - samples
+      summary: Search samples using a Mongo query
+      description: ''
+      operationId: postSample
+      requestBody:
+        description: Sample object that needs to be added
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+      responses:
+        '200':
+          description: successful operation
+        '405':
+          description: Invalid input
+components:
+  schemas:
+    Sample:
```

Comment:
This looks like a YAML transcription of https://github.com/NSLS-II/amostra/blob/master/amostra/schemas/sample.json. It looks like it _may_ be possible to refer to the original file directly instead of storing a (JSON-to-YAML translated) copy here. The advantage of this approach would be that we'd only have to make updates in one place, and we wouldn't need to worry about them getting accidentally out of sync.

Possibly useful link: https://swagger.io/docs/specification/using-ref/